allennlp_models/generation/models/bart.py

```diff
@@ -263,29 +264,28 @@ def forward(
         return outputs
 
     @staticmethod
-    def _decoder_cache_to_dict(decoder_cache):
+    def _decoder_cache_to_dict(decoder_cache: DecoderCacheType) -> Dict[str, torch.Tensor]:
         cache_dict = {}
         for layer_index, layer_cache in enumerate(decoder_cache):
-            for attention_name, attention_cache in layer_cache.items():
-                for tensor_name, cache_value in attention_cache.items():
-                    key = (layer_index, attention_name, tensor_name)
-                    cache_dict[key] = cache_value
+            assert len(layer_cache) == 4
```

Comment:
Why are there four layers? I assume this is related to the internal huggingface changes?

Reply:
Yea, this is internal to HF. Each layer cache contains the key and value tensors for the self-attention and cross-attention. So 4 tensors per layer. I'll add a comment on that.